core/src/main/scala/org/bykn/bosatsu/TypeRef.scala

```diff
@@ -58,7 +58,14 @@ object TypeRef {
     def toBoundVar: Type.Var.Bound = Type.Var.Bound(asString)
   }
   case class TypeName(name: Name) extends TypeRef
-  case class TypeArrow(from: TypeRef, to: TypeRef) extends TypeRef
+  case class TypeArrow(from: NonEmptyList[TypeRef], to: TypeRef) extends TypeRef
+
+  object TypeArrow {
+    // the common case of Fn1
```

Comment:
remove this and move it to the tests or give it a name like `TypeArrow.oneArg` or something so it isn't confused with the default apply.